python-checks/src/main/java/org/sonar/python/checks/TorchLoadLeadsToUntrustedCodeExecutionCheck.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.Tree;
+
+@Rule(key = "S6985")
+public class TorchLoadLeadsToUntrustedCodeExecutionCheck extends PythonSubscriptionCheck {
+
+  public static final String TORCH_LOAD = "torch.load";
+  public static final String MESSAGE = "Replace this call with a safe alternative";
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, ctx -> {
+      CallExpression callExpression = (CallExpression) ctx.syntaxNode();
+      Symbol calleeSymbol = callExpression.calleeSymbol();
+      if(calleeSymbol != null && TORCH_LOAD.equals(calleeSymbol.fullyQualifiedName())) {
```

Comment:
`calleeSymbol != null` is missing test coverage. However, this is hard to test for and would complicate the tests unnecessarily

Reply:
Are you sure? If you search for `calleeSymbol != null` within the project, this is a construct that is used many times and is most likely tested somewhere else. I think ensuring this is well-tested should not be too complex.

Reply:
It would require something like `(lambda x: x)()` in the lit file, at least as far as I understand it. Or is there a test case which makes sense test where this can occure?

Reply:
To cover this, you only have to have a case that triggers it in the unit test file of the rule (i.e `torchLoadLeadsToUntrustedCodeExecution.py`). Callees can be missing symbols for many reasons, indeed as an immediately called lambda, or probably even something simpler like `something[42]()`. There are most likely many other cases that could trigger this as well.